azure-maven-plugin-lib/src/main/java/com/microsoft/azure/common/project/IProject.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.common.project;
+
+import java.nio.file.Path;
+import java.util.Collection;
+
+public interface IProject {
+    Path getBaseDirectory();
+
+    Path getArtifactFile();
+
+    Path getBuildDirectory();
+
+    Path getClassesOutputDirectory();
```

Comment:
`getClassesOutputDirectory` seems could be removed as common library will not do jobs depends on classes